src/frameworks/next/index.ts

```diff
@@ -300,14 +331,17 @@ export async function ɵcodegenFunctionsDirectory(sourceDir: string, destDir: st
 /**
  * Create a dev server.
  */
-export async function getDevModeHandle(dir: string) {
+export async function getDevModeHandle(dir: string, hostingEmulatorInfo?: EmulatorInfo) {
   const { default: next } = relativeRequire(dir, "next");
   const nextApp = next({
     dev: true,
     dir,
+    hostname: hostingEmulatorInfo?.host,
+    port: hostingEmulatorInfo?.port,
   });
   const handler = nextApp.getRequestHandler();
   await nextApp.prepare();
+  // TODO can we check for middleware and error if we don't have hostingEmulatorInfo
```

Comment:
I wasn't able to find a good way to hook into `firebase serve` and get the port number. IMO we're steering away from serve anyway, so let's punt. Taking a TODO to see if we can error but we shouldn't hold this up

Reply:
The manifest files (in dev mode) don't have middleware, we could potentially just look for a middleware.js/ts file & warn